In [3]:
#Importing necessary libraries
import pandas as pd
import numpy as np

In [4]:
#Loading the datasets for data cleaning
data=pd.read_csv("C:\\Users\\USER\\OneDrive\\Documents\\Social Buzz Project\\Content.csv")

In [5]:
data.head()

Unnamed: 0                            Content ID  \
0           0  97522e57-d9ab-4bd6-97bf-c24d952602d2   
1           1  9f737e0a-3cdd-4d29-9d24-753f4e3be810   
2           2  230c4e4d-70c3-461d-b42c-ec09396efb3f   
3           3  356fff80-da4d-4785-9f43-bc1261031dc6   
4           4  01ab84dd-6364-4236-abbb-3f237db77180   

                                User ID   Type        Category  \
0  8d3cd87d-8a31-4935-9a4f-b319bfe05f31  photo        Studying   
1  beb1f34e-7870-46d6-9fc7-2e12eb83ce43  photo  healthy eating   
2  a5c65404-5894-4b87-82f2-d787cbee86b4  photo  healthy eating   
3  9fb4ce88-fac1-406c-8544-1a899cee7aaf  photo      technology   
4  e206e31b-5f85-4964-b6ea-d7ee5324def1  video            food   

                                                 URL  
0  https://socialbuzz.cdn.com/content/storage/975...  
1  https://socialbuzz.cdn.com/content/storage/9f7...  
2  https://socialbuzz.cdn.com/content/storage/230...  
3  https://socialbuzz.cdn.com/content/storage/356...  
4  https://socialbuzz.cdn.com/content/storage/01a...

In [6]:
#Checking for number of columns and rows
no_of_rows=data.shape[0]
no_of_columns=data.shape[1]
print(no_of_rows)
print(no_of_columns)

1000
6


The content dataset has 1000 rows and 6 columns.

In [8]:
#DATA CLEANING 
#HANDLING MISSING VALUES
#general view of how many missing values there is per column 
missing_values_count=data.isnull().sum()
missing_values_count[0:6]

Unnamed: 0      0
Content ID      0
User ID         0
Type            0
Category        0
URL           199
dtype: int64

The url column has 199 missing values
We are going to drop the missing values in the URL column.

In [9]:
#dropping the entire URL column using the inplace=true function
data.dropna(subset=["URL"],inplace=True)

In [11]:
#Changing the dataheads to snake_case
data.columns=data.columns.str.replace(" ","_")
print(data.columns)

Index(['Unnamed:_0', 'Content_ID', 'User_ID', 'Type', 'Category', 'URL'], dtype='object')


In [26]:
#identifying the rows that have incorrect data using boolean indexing
column_name = "Category"
# Use the str.replace() method to replace values case-insensitively
data[column_name] = data[column_name].str.replace("Studying", "studying", case=False)
data[column_name] = data[column_name].str.replace("Education", "education", case=False)
data[column_name] = data[column_name].str.replace("Healthy Eating", "healthy eating", case=False)
data[column_name] = data[column_name].str.replace("Science", "science", case=False)
data[column_name] = data[column_name].str.replace("Animals", "animals", case=False)
data[column_name] = data[column_name].str.replace("Veganism", "veganism", case=False)
data[column_name] = data[column_name].str.replace("Fitness", "fitness", case=False)
data[column_name] = data[column_name].str.replace("Technology", "technology", case=False)
data[column_name] = data[column_name].str.replace("Soccer", "soccer", case=False)
data[column_name] = data[column_name].str.replace("Culture", "culture", case=False)
data[column_name] = data[column_name].str.replace("Travel", "travel", case=False)

In [27]:
#Downloading the cleaned dataset
excel_file_path=r"C:\\Users\\USER\\Downloads\\Content.xlsx"
data.to_excel(excel_file_path, index=False)

In [28]:
#CLEANING DATASET 2
data=pd.read_csv("C:\\Users\\USER\\OneDrive\\Documents\\Accenture Project\\Reactions.csv")

In [29]:
data.head()

Unnamed: 0                            Content ID  \
0           0  97522e57-d9ab-4bd6-97bf-c24d952602d2   
1           2  97522e57-d9ab-4bd6-97bf-c24d952602d2   
2           3  97522e57-d9ab-4bd6-97bf-c24d952602d2   
3           4  97522e57-d9ab-4bd6-97bf-c24d952602d2   
4           5  97522e57-d9ab-4bd6-97bf-c24d952602d2   

                                User ID        Type          Datetime  
0                                   NaN         NaN  22/04/2021 15:17  
1  92b87fa5-f271-43e0-af66-84fac21052e6     dislike  17/06/2021 12:22  
2  163daa38-8b77-48c9-9af6-37a6c1447ac2      scared  18/04/2021 05:13  
3  34e8add9-0206-47fd-a501-037b994650a2     disgust  06/01/2021 19:13  
4  9b6d35f9-5e15-4cd0-a8d7-b1f3340e02c4  interested  23/08/2020 12:25

This dataset has some missing values as shown in the 5 rows displayed above.

In [30]:
#checking the datatypes in this dataset
data.dtypes

Unnamed: 0     int64
Content ID    object
User ID       object
Type          object
Datetime      object
dtype: object

In [31]:
#checking the number of columns and rows in this dataset
no_of_rows=data.shape[0]
no_of_columns=data.shape[1]
print(no_of_rows)
print(no_of_columns)

25552
5


This dataset has 25,553 rows and 5 columns

In [32]:
#Changing the dataheads to snake_case
data.columns=data.columns.str.replace(" ","_")
print(data.columns)

Index(['Unnamed:_0', 'Content_ID', 'User_ID', 'Type', 'Datetime'], dtype='object')


In [33]:
#checking the missing values in each column
missing_values_count=data.isnull().sum()
missing_values_count[0:5]

Unnamed:_0       0
Content_ID       0
User_ID       3019
Type           980
Datetime         0
dtype: int64

The user ID and the Type ID has missing values

In [34]:
#Dropping the missing values in the two columns
data.dropna(subset=["User_ID","Type"],inplace=True)

In [35]:
from datetime import datetime
column_name ="Datetime" 
# Specify the date format
date_format = "%d/%m/%Y %H:%M"
# Use the apply method to convert the entire column to datetime
data[column_name] = data[column_name].apply(lambda x: datetime.strptime(x, date_format))


In [36]:
#Downloading the cleaned dataset
excel_file_path=r"C:\\Users\\USER\\Downloads\\Reactions.xlsx"
data.to_excel(excel_file_path, index=False)

In [37]:
#CLEANING DATASET 3
data=pd.read_csv("C:\\Users\\USER\\OneDrive\\Documents\\Accenture Project\\ReactionTypes.csv")

In [38]:
data.head()

Unnamed: 0        Type Sentiment  Score
0           0       heart  positive     60
1           1        want  positive     70
2           2     disgust  negative      0
3           3        hate  negative      5
4           4  interested  positive     30

In [39]:
#Checking for number of columns
no_of_rows=data.shape[0]
no_of_columns=data.shape[1]
print(no_of_rows)
print(no_of_columns)

16
4


This dataset has 16 rows and 4 columns.

In [40]:
data.describe()

Unnamed: 0      Score
count   16.000000  16.000000
mean     7.500000  39.625000
std      4.760952  26.901983
min      0.000000   0.000000
25%      3.750000  14.250000
50%      7.500000  40.000000
75%     11.250000  66.250000
max     15.000000  75.000000

In [41]:
#checking the datatypes
data.dtypes

Unnamed: 0     int64
Type          object
Sentiment     object
Score          int64
dtype: object

In [42]:
#Changing the dataheads to snake_case
data.columns=data.columns.str.replace(" ","_")
print(data.columns)

Index(['Unnamed:_0', 'Type', 'Sentiment', 'Score'], dtype='object')


In [43]:
excel_file_path=r"C:\\Users\\USER\\Downloads\\Reaction Types.xlsx"
data.to_excel(excel_file_path, index=False)

In [36]:
#DATA MODELLING
#LOADING THE CLEANED DATASETS AS NEW DATAFRAMES
import pandas as pd
import numpy as numpy
Content_data=pd.read_csv("Content.csv")
Reaction_Types_data=pd.read_csv("ReactionTypes.csv")
Reaction_data=pd.read_csv("Reactions.csv")


#merging the  three datasets using the MERGE() FUNCTION 
#based on the common column,that is "content_id"
#merged_data=Reaction_data.merge(Content_data, on="Content_ID", how="inner"copy=
#step 3: Join reaction with content using Content ID as the common key
merged_data = pd.merge(Reaction_data, Content_data, left_on="Content ID", right_on="Content ID", how="inner")
 
print(merged_data.columns)
#print("Reaction Types Data Columns:", Reaction_Types_data.columns)
#Step 3: Join the result from step 2 with reaction_type_data using the type column
final_data = pd.merge(merged_data, Reaction_Types_data, left_on= "Type_x",right_on="Type",how='inner')
print(final_data)

Index(['Unnamed: 0_x', 'Content ID', 'User ID_x', 'Type_x', 'Datetime',
       'Unnamed: 0_y', 'User ID_y', 'Type_y', 'Category', 'URL'],
      dtype='object')
       Unnamed: 0_x                            Content ID  \
0                 1  97522e57-d9ab-4bd6-97bf-c24d952602d2   
1                 4  97522e57-d9ab-4bd6-97bf-c24d952602d2   
2                35  97522e57-d9ab-4bd6-97bf-c24d952602d2   
3                52  9f737e0a-3cdd-4d29-9d24-753f4e3be810   
4                88  230c4e4d-70c3-461d-b42c-ec09396efb3f   
...             ...                                   ...   
24568         25489  435007a5-6261-4d8b-b0a4-55fdc189754b   
24569         25491  435007a5-6261-4d8b-b0a4-55fdc189754b   
24570         25512  4e4c9690-c013-4ee7-9e66-943d8cbd27b7   
24571         25524  4e4c9690-c013-4ee7-9e66-943d8cbd27b7   
24572         25531  4e4c9690-c013-4ee7-9e66-943d8cbd27b7   

                                  User ID_x   Type_x             Datetime  \
0      5d454588-283d-459d-915d

In [37]:
final_data

Unnamed: 0_x                            Content ID  \
0                 1  97522e57-d9ab-4bd6-97bf-c24d952602d2   
1                 4  97522e57-d9ab-4bd6-97bf-c24d952602d2   
2                35  97522e57-d9ab-4bd6-97bf-c24d952602d2   
3                52  9f737e0a-3cdd-4d29-9d24-753f4e3be810   
4                88  230c4e4d-70c3-461d-b42c-ec09396efb3f   
...             ...                                   ...   
24568         25489  435007a5-6261-4d8b-b0a4-55fdc189754b   
24569         25491  435007a5-6261-4d8b-b0a4-55fdc189754b   
24570         25512  4e4c9690-c013-4ee7-9e66-943d8cbd27b7   
24571         25524  4e4c9690-c013-4ee7-9e66-943d8cbd27b7   
24572         25531  4e4c9690-c013-4ee7-9e66-943d8cbd27b7   

                                  User ID_x   Type_x             Datetime  \
0      5d454588-283d-459d-915d-c48a2cb4c27f  disgust  2020-11-07 09:43:50   
1      34e8add9-0206-47fd-a501-037b994650a2  disgust  2021-01-06 19:13:01   
2      13c06e7e-833d-47eb-a790-5e09ccfd8d2c  disgust  2021-04-09 02:46:20   
3      8b49caad-bcc5-43de-bf40-34a66ff8805c  disgust  2021-03-28 21:15:26   
4      ef147ea5-9696-44d5-b6c2-a43f62fd8ce2  disgust  2020-08-04 05:40:33   
...                                     ...      ...                  ...   
24568  d51a3403-8bab-4d88-af4f-a429c6d8c168    adore  2020-10-04 22:26:33   
24569  3663e3e6-3d5c-4ed9-a6af-1e680ec5f34b    adore  2020-09-18 10:50:50   
24570  89572273-761e-46f3-8b99-1689d98aac90    adore  2020-10-31 03:58:44   
24571  7bf3ce80-784e-42a0-b06a-dbbd1e31a9b7    adore  2020-06-25 15:12:29   
24572  dcc6000c-ba92-472c-8bc1-9e9ca30b475b    adore  2020-12-17 16:32:57   

       Unnamed: 0_y                             User ID_y Type_y  \
0                 0  8d3cd87d-8a31-4935-9a4f-b319bfe05f31  photo   
1                 0  8d3cd87d-8a31-4935-9a4f-b319bfe05f31  photo   
2                 0  8d3cd87d-8a31-4935-9a4f-b319bfe05f31  photo   
3                 1  beb1f34e-7870-46d6-9fc7-2e12eb83ce43  photo   
4                 2  a5c65404-5894-4b87-82f2-d787cbee86b4  photo   
...             ...                                   ...    ...   
24568           997  35d6a1f3-e358-4d4b-8074-05f3b7f35c2a  audio   
24569           997  35d6a1f3-e358-4d4b-8074-05f3b7f35c2a  audio   
24570           998  b9bcd994-f000-4f6b-87fc-caae08acfaa1    GIF   
24571           998  b9bcd994-f000-4f6b-87fc-caae08acfaa1    GIF   
24572           998  b9bcd994-f000-4f6b-87fc-caae08acfaa1    GIF   

             Category                                                URL  \
0            Studying  https://socialbuzz.cdn.com/content/storage/975...   
1            Studying  https://socialbuzz.cdn.com/content/storage/975...   
2            Studying  https://socialbuzz.cdn.com/content/storage/975...   
3      healthy eating  https://socialbuzz.cdn.com/content/storage/9f7...   
4      healthy eating  https://socialbuzz.cdn.com/content/storage/230...   
...               ...                                                ...   
24568        veganism  https://socialbuzz.cdn.com/content/storage/435...   
24569        veganism  https://socialbuzz.cdn.com/content/storage/435...   
24570         culture  https://socialbuzz.cdn.com/content/storage/4e4...   
24571         culture  https://socialbuzz.cdn.com/content/storage/4e4...   
24572         culture  https://socialbuzz.cdn.com/content/storage/4e4...   

       Unnamed: 0     Type Sentiment  Score  
0               2  disgust  negative      0  
1               2  disgust  negative      0  
2               2  disgust  negative      0  
3               2  disgust  negative      0  
4               2  disgust  negative      0  
...           ...      ...       ...    ...  
24568           9    adore  positive     72  
24569           9    adore  positive     72  
24570           9    adore  positive     72  
24571           9    adore  positive     72  
24572           9    adore  positive     72  

[24573 rows x 14 columns]

In [44]:
#Using excel write to create two sheets in one spreadsheet
with pd.ExcelWriter("Final Data.xlsx") as excel:
    
    final_data.to_excel(excel,index=False,sheet_name="final_data")
    final_data["Category"].value_counts()[:5].reset_index().to_excel(excel,index=False,sheet_name="top_5_categories")

